admin-ui/plugins/admin/components/MAU/MauGraph.js

```diff
@@ -9,35 +9,49 @@ import GluuViewWrapper from 'Routes/Apps/Gluu/GluuViewWrapper'
 import { getMau } from 'Plugins/admin/redux/features/mauSlice'
 import GluuLabel from 'Routes/Apps/Gluu/GluuLabel'
 import { Button, Card, CardFooter, CardBody, FormGroup, Col, Row } from 'Components'
-import { hasBoth, buildPayload, STAT_READ, STAT_JANS_READ } from 'Utils/PermChecker'
+import { buildPayload, STAT_READ, STAT_JANS_READ } from 'Utils/PermChecker'
 import { useTranslation } from 'react-i18next'
 import { useDispatch, useSelector } from 'react-redux'
 import SetTitle from 'Utils/SetTitle'
 import applicationStyle from 'Routes/Apps/Gluu/styles/applicationstyle'
 import { ThemeContext } from 'Context/theme/themeContext'
 import dayjs from 'dayjs'
+import { useCedarling } from '@/cedarling'
 
 const ActiveUsersGraph = lazy(() => import('Routes/Dashboards/Grapths/ActiveUsersGraph'))
 
 function MauGraph() {
   const dispatch = useDispatch()
   const statData = useSelector((state) => state.mauReducer.stat)
   const loading = useSelector((state) => state.mauReducer.loading)
-  const permissions = useSelector((state) => state.authReducer.permissions)
+  const { permissions: cedarPermissions } = useSelector((state) => state.cedarPermissions)
   const { t } = useTranslation()
   const theme = useContext(ThemeContext)
   const selectedTheme = theme.state.theme
   const [startDate, setStartDate] = useState(dayjs().subtract(3, 'months'))
   const [endDate, setEndDate] = useState(dayjs())
+  const { hasCedarPermission, authorize } = useCedarling()
   const userAction = {}
   const options = {}
+  const initPermissions = async () => {
+    const permissions = [STAT_READ, STAT_JANS_READ]
+    for (const permission of permissions) {
+      await authorize([permission])
+    }
+  }
 
   useEffect(() => {
     if (statData.length === 0 || !statData) {
       search()
     }
+    initPermissions()
   }, [])
```

Comment:
Null-guard order can throw when statData is undefined.

Accessing statData.length before checking truthiness risks a crash.

Apply:

```diff
-  useEffect(() => {
-    if (statData.length === 0 || !statData) {
+  useEffect(() => {
+    if (!statData || statData.length === 0) {
       search()
     }
-    initPermissions()
-  }, [])
+    initPermissions()
+  }, [])
```